### PR TITLE
Add session expiration middleware

### PR DESCRIPTION
**Pull request by Ian-MacLeod**
If we want sessions to expire on browser close, we cannot use cookie expiration to ensure that sessions expire after a certain amount of time. This adds a middleware that can be used to expire sessions after a period of user inactivity.

### CODE REVIEW

File: docs/usage/middleware.rst [hunk 1]
```diff
@@ -11,3 +14,27 @@ Add it as the first entry in your ``MIDDLEWARE`` list in your settings file:
...
+       # ...
+   ]
+
+To customize the behavior of this middleware, the following settings are available:
+
+``SESSION_EXPIRATION_SECONDS`` - how long to wait in seconds before expiring the user's session. Default: 3600.
+
+``SESSION_EXPIRATION_ACTIVITY_RESETS`` - boolean that determines whether user activity delays the expiration of the session. Default: True.
+
+``SESSION_EXIRATION_KEY`` - key to use in the session to track the expiration time. Default: "_last_active_at".
```
Comment (cviveiro) [inline docs/usage/middleware.rst:40]: `EXPIRATION`

File: zygoat_django/middleware/session_expiration.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import time
+
+from django.conf import settings
+
```
Comment (cviveiro) [inline zygoat_django/middleware/session_expiration.py:4]: needs another blank line between imports and constants

File: zygoat_django/middleware/session_expiration.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+SESSION_EXPIRATION_SECONDS = getattr(settings, "SESSION_EXPIRATION_SECONDS", 3600)
+SESSION_EXPIRATION_ACTIVITY_RESETS = getattr(
+    settings, "SESSION_EXPIRATION_ACTIVITY_RESETS", True
+)
+SESSION_EXIRATION_KEY = getattr(settings, "SESSION_EXIRATION_KEY", "_last_active_at")
```
Comment (cviveiro) [inline zygoat_django/middleware/session_expiration.py:9]: ahhh so it is consistent, but yeah, should be corrected.

File: zygoat_django/middleware/session_expiration.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    def middleware(request):
+        if has_session(request):
+            last_activity = request.session.get(SESSION_EXIRATION_KEY)
+            if (
+                last_activity is None
+                or time.time() - last_activity > SESSION_EXPIRATION_SECONDS
+            ):
+                request.session.flush()
+
+        response = get_response(request)
```
Comment (cviveiro) [inline zygoat_django/middleware/session_expiration.py:30]: interesting, middleware really is just a wrapper! til, never looked at middleware sourcecode before

**Review by cviveiro — COMMENTED** [review]
how does this relate to https://github.com/MetLifeLegalPlans/django-willing-zg/pull/5?

haven't looked at the test file yet, will try to get to that soon

**Comment by Ian-MacLeod** [comment]
Thanks for the comments, updated!

**Pull request merged**